packages/flutter/test/material/mergeable_material_test.dart

```diff
@@ -0,0 +1,170 @@
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(box.size.height, lessThan(332));
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(box.size.height, equals(332));
+
+    // Remove one gap.
+    await tester.pumpWidget(
+      new Scaffold(
+        body: new ScrollableViewport(
+          child: new MergeableMaterial(
+            key: mmKey,
+            children: <MergeableMaterialItem>[
+              new MaterialSlice(
+                key: sliceKey,
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(box.size.height, equals(316));
+
+    // Add two slices.
+    await tester.pumpWidget(
+      new Scaffold(
+        body: new ScrollableViewport(
+          child: new MergeableMaterial(
+            key: mmKey,
+            children: <MergeableMaterialItem>[
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                key: sliceKey,
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
```

Comment:
I'm not so sure how to verify changing elevation. Would getting a `RenderObject` and painting it with a mocked `Canvas` *after* and elevation change be enough? (I'm guessing that it doesn't test properly since is doesn't guarantee that `updateRenderObject` actually gets called)